ci/autoresearch/phases.py

```
@@ -773,12 +773,56 @@ def _parse_args_and_build_commands(args: Args) -> RunContext | int:
+    elif final_environment:
+        # Board known up-front \u2014 synthesise now so downstream callers (chip
+        # auto-detect log lines, default_envs probes) see the staged file.
+        from ci.autoresearch.staging import synthesise_autoresearch_project
+
+        try:
+            build_dir = synthesise_autoresearch_project(
+                final_environment,
+                project_root=project_root,
+                verbose=args.verbose,
+            )
+        except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)